src/Azure.Functions.Cli/Diagnostics/ColoredConsoleLogger.cs

```diff
@@ -5,50 +5,74 @@
     {
-        private readonly Func<string, LogLevel, bool> _filter;
         private readonly bool _verboseErrors;
         private readonly string _category;
+        private readonly LoggingFilterHelper _loggingFilterOptions;
+        private readonly string[] allowedLogsPrefixes = new string[] { "Worker process started and initialized.", "Host lock lease acquired by instance ID" };
 
-        public ColoredConsoleLogger(string category, Func<string, LogLevel, bool> filter = null)
+        public ColoredConsoleLogger(string category, LoggingFilterHelper loggingFilterOptions)
         {
             _category = category;
-            _filter = filter;
+            _loggingFilterOptions = loggingFilterOptions;
             _verboseErrors = StaticSettings.IsDebug;
         }
 
         public bool IsEnabled(LogLevel logLevel)
         {
-            if (_filter == null)
-            {
-                return true;
-            }
-
-            return _filter(_category, logLevel);
+            return _loggingFilterOptions.IsEnabled(_category, logLevel);
         }
 
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
-            if (!IsEnabled(logLevel))
+            string formattedMessage = formatter(state, exception);
+
+            if (string.IsNullOrEmpty(formattedMessage))
             {
                 return;
             }
 
-            string formattedMessage = formatter(state, exception);
+            if (DoesMessageStartsWithAllowedLogsPrefix(formattedMessage))
+            {
+                LogToConsole(logLevel, exception, formattedMessage);
+                return;
+            }
 
-            if (string.IsNullOrEmpty(formattedMessage))
+            if (!IsEnabled(logLevel))
             {
                 return;
             }
 
+            LogToConsole(logLevel, exception, formattedMessage);
+        }
+
+        private void LogToConsole(LogLevel logLevel, Exception exception, string formattedMessage)
+        {
             foreach (var line in GetMessageString(logLevel, formattedMessage, exception))
             {
-                ColoredConsole.WriteLine($"[{DateTime.UtcNow}] {line}");
+                var outputline = $"{line}";
+                if (_loggingFilterOptions.VerboseLogging)
+                {
+                    outputline = $"[{DateTime.UtcNow}] {outputline}";
+                }
+                ColoredConsole.WriteLine($"{outputline}");
+            }
+        }
+
+        internal bool DoesMessageStartsWithAllowedLogsPrefix(string formattedMessage)
+        {
+            if (formattedMessage == null)
+            {
+                throw new ArgumentNullException(nameof(formattedMessage));
             }
+            var formattedMessagesWithWhiteListePrefixes = allowedLogsPrefixes.Where(s => formattedMessage.StartsWith(s, StringComparison.OrdinalIgnoreCase));
```

Comment:
use `AllowList`

Reply:
Actually -- you can `return allowedLogsPrefixes.Any(s => formattedMessage.StartsWith(s, StringComparison.OrdinalIgnoreCase));`...